edx_sga/tests/common.py

```diff
@@ -71,3 +72,20 @@ def parse_timestamp(timestamp):
     return datetime.strptime(timestamp, DateTime.DATETIME_FORMAT).replace(
         tzinfo=pytz.utc
     )
+
+
+def reformat_xml(xml_string):
+    """
+    Parse and render the XML to remove whitespace differences
+    """
+    parser = etree.XMLParser(remove_blank_text=True)
+    root = etree.XML(xml_string, parser=parser)
+
+    # Remove whitespace
+    for elem in root.iter('*'):
+        if elem.text is not None:
+            elem.text = elem.text.strip()
+            if not elem.text:
+                elem.text = None
+
+    return etree.tostring(root)
```

Comment:
Sorry, just noticed this. I ran this code and the two parsed variables were equal. I don't think you need this whitespace-stripping logic. Seems like you can use the logic below and call this function something like `simplify_xml`

```python
from lxml import etree

xml1 = '<this><is some="xml" /></this>'
xml2 = """
<this>   
  <is some="xml" /> 
</this>
"""

parser = etree.XMLParser(remove_blank_text=True)
parsed1 = etree.XML(xml1, parser=parser)
parsed2 = etree.XML(xml2, parser=parser)

etree.tostring(parsed1) == etree.tostring(parsed2)
```

Reply:
That will remove whitespace around the `<is ... />` tag but it won't trim whitespace in the child node if the parser thinks it's significant:

    >>> xml3 = '<this> <is some="xml" > </is> </this>'
    >>> parsed3 = etree.XML(xml3, parser=parser)
    >>> etree.tostring(parsed3)
    b'<this><is some="xml"> </is></this>'

Reply:
aaaah understood